cpp/src/io/parquet/writer_impl.cu

```diff
@@ -792,6 +810,40 @@ void writer::write_chunked_end(pq_chunked_state &state){
   _impl->write_chunked_end(state);
 }
 
+std::vector<uint8_t> writer::merge_rowgroup_metadata(const std::vector<const std::vector<uint8_t>*>& metadata_list)
```

Comment:
`raw_metadata_out` byte-arrays will be stored on host, correct?

I am assuming it will be straightforward cython to enable a dask client to store a list of these byte-arrays in python (with each element created during the write call), and then pass them in to this function.  @kkraus14 - Please do let me know if I am failing to consider some complication.

Reply:
Yup these are all host vectors and it's pretty straightforward to handle them in Cython and even bring them into Python land zero copy.

Reply:
Yes, it's basically a big string, I'm considering changing this to be an array of vectors rather than an array of pointers to vectors to make it simpler on the cython layer, though this would force a copy. From what I understand, the copy is pretty much needed anyways due to c vs python ownership, unless I change this to be a vector of raw {constptr, length} pairs (the length of each metadata chunk is typically in the range of ~40 bytes per column per rowgroup, so we're not talking tremendous amounts compared to the dataframe itself)

Reply:
Could you make it a vector of smart pointers instead? That makes it easy to handle the ownership semantics between cpp and cython while allowing us to use move semantics instead of having to make a copy.

Reply:
Sure can.

Reply:
@kkraus14 To be clear, you'd want a vector of unique_ptr<vector<uint8_t>> ?

Reply:
Got the PR ready on the C++ side using unique_ptr[vector[uint8_t]] for both input & output, but now scratching my head about how to go from unique_ptr[vector[uint8_t]] to a python bytes object (ideally with zero copy, but I have yet to figure out a way that just compiles :) ).

Reply:
@OlivierNV you should be able to do something like:
```
cdef unique_ptr[vector[uint8_t]] cython_obj = move(c_obj) # May need to add a move declaration to the Cython  in `moves.pxd` to handle this
cdef uint8_t[:] cython_array = <uint8_t[:]>c_obj.get()
```

Now if we want this memory to stay alive outside of the Cython scope we'd have to either build a wrapper Cython class to keep the smart pointer alive, or release the smart pointer and use something like a Cython array (https://cython.readthedocs.io/en/latest/src/userguide/memoryviews.html#cython-arrays) to manage the memory lifetime.

Reply:
Tentatively marking the PR as ready for review, but I think if we want zero copy, the merge api should take in a vector of {ptr, bytecount} pairs (otherwise the use of a uint8 vector implies ownership by the cpp layer)

Reply:
Handling a `std::vector<std::pair<...>>` efficiently from the Python / Cython side would be a bit of a pain. I would much reather keep it as a `std::vector<uint8_t>` which we can naturally bridge into the Python world zero copy while using C++ smart pointer lifetime semantics.

Reply:
I meant for the input to merge, not the return (currently a vector of unique_ptr of uint8_t vector), as the input is likely to be created on the python side whereas a uint8_t vector implies cpp ownership (no lifetime issues here since it's only used during the call to the cpp layer). But if it's no big deal to create c++ vectors from python land, then it's fine as is.